### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Limits/Shapes/Biproducts): add associator

### PR DESCRIPTION
**Pull request by eric-wieser**
As far as I can tell this doesn't exist anywhere else. The motivation is #7125, but I have a lot less idea of what I'm doing there than I do here.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/Shapes/Biproducts.lean [hunk 1]
```diff
@@ -1960,6 +1957,26 @@ theorem biprod.symmetry (P Q : C) : (biprod.braiding P Q).hom ≫ (biprod.braidi
...
 
+/-- The associator isomorphism which associates a binary biproduct. -/
+@[simps]
+def biprod.associator (P Q R : C) : (P ⊞ Q) ⊞ R ≅ P ⊞ (Q ⊞ R)  where
+  hom := biprod.lift (biprod.fst ≫ biprod.fst) (biprod.lift (biprod.fst ≫ biprod.snd) biprod.snd)
+  inv := biprod.lift (biprod.lift biprod.fst (biprod.snd ≫ biprod.fst)) (biprod.snd ≫ biprod.snd)
+
+/-- The associator isomorphism can be passed through a map by swapping the order. -/
+@[reassoc]
+theorem biprod.associator_natural {U V W X Y Z : C} (f : U ⟶ X) (g : V ⟶ Y) (h : W ⟶ Z) :
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Limits/Shapes/Biproducts.lean:1968]: `biprod.associator_hom_naturality` might be a slightly better name; and `biprod.associator_inv_natural(ity)` also.

**Review by joelriou — APPROVED** [review]

**Comment by joelriou** [comment]
bors d+

File: Mathlib/CategoryTheory/Limits/Shapes/Biproducts.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/CategoryTheory/Limits/Shapes/Biproducts.lean:1968]: This name was copied from `biprod.braid_natural` above. Would you like me to rename both, or are you happy for me to leave things as is?

Comment (joelriou) [reply Mathlib/CategoryTheory/Limits/Shapes/Biproducts.lean:1968]: Ah ok. Then, it is ok to leave it as is.

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**